ddp-workspace/projects/ddp-rarex/src/app/services/participant.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { BehaviorSubject, Observable, Subject } from 'rxjs';
+import { filter, map, switchMap, takeUntil } from 'rxjs/operators';
+
+import {
+  SessionMementoService,
+  UserProfile,
+  UserProfileServiceAgent,
+} from 'ddp-sdk';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ParticipantService implements OnDestroy {
+  private destroyed = new Subject<void>();
+  private currentParticipantGuid$ = new BehaviorSubject<string | null>(null);
+  private prevParticipantGuid$ = new BehaviorSubject<string | null>(null);
+  private currentParticipantProfile$ = new BehaviorSubject<UserProfile | null>(
+    null,
+  );
+  private sessionSub = this.sessionService.sessionObservable
+    .pipe(
+      takeUntil(this.destroyed),
+      filter(session => !!session),
+      switchMap(session =>
+        this.userProfileService.profile.pipe(
```

Comment:
Does this mean we only fetch/store the profile when we have a session (e.g. logged in), so this is effectively static? For RareX, the name gets populated in the backend after they submit the consent. Does this handle "refreshing" the profile to fetch the new name?